File: Singleton/Singleton.py
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
            cls._instance.initialize()
        return cls._instance
    
    def initialize(self):
        self.value = "Hello World"

    @classmethod
    def get_instance(cls):
        if cls._instance is None:
            cls._instance = cls()
        return cls._instance

if __name__ == "__main__":
    instance1 = Singleton.get_instance()
    print(f"Singleton: {instance1.value}")

File: Singleton/Singleton.java
package Singleton;

public class Singleton {
    
    private static Singleton singleton;

    public static synchronized Singleton getInstance() {
        if (singleton == null) {
            singleton = new Singleton();
        }
        return singleton;
    }

    public String showMessage() {
        return "Hello world!";
    }

}

